src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerBulkData.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.sqlserver.jdbc;
+
+import java.io.Serializable;
+
+/**
+ * Provides an interface used to create classes that read in data from any source (such as a file) and allows a
+ * SQLServerBulkCopy class to write the data to SQL Server tables.
+ */
+public interface ISQLServerBulkData extends Serializable {
+
+    /**
+     * Returns the ordinals for each of the columns represented in this data record.
+     *
+     * @return Set of ordinals for the columns.
+     */
+    java.util.Set<Integer> getColumnOrdinals();
+
+    /**
+     * Returns the name of the given column.
+     *
+     * @param column
+     *        Column ordinal
+     * @return Name of the column
+     */
+    String getColumnName(int column);
+
+    /**
+     * Returns the JDBC data type of the given column.
+     *
+     * @param column
+     *        Column ordinal
+     * @return JDBC data type of the column
+     */
+    int getColumnType(int column);
+
+    /**
+     * Returns the precision for the given column.
+     *
+     * @param column
+     *        Column ordinal
+     * @return Precision of the column
+     */
+    int getPrecision(int column);
+
+    /**
+     * Returns the scale for the given column.
+     *
+     * @param column
+     *        Column ordinal
+     * @return Scale of the column
+     */
+    int getScale(int column);
+
+    /**
+     * Returns whether the column represents an identity column.
+     *
+     * @param column
+     *        Column ordinal
+     * @return True if the column is an identity column; false otherwise.
+     */
+    boolean isAutoIncrement(int column);

```

Comment:
We don't use isAutoIncrement in writeToServer, so I think we can leave this method back in ISQLServerBulkRecord.

Reply:
I'll move this back